Gridy/View/Position.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  Position.swift
+//  Gridy
+//
+//  Created by Alexandra Ivanova on 11/10/2020.
+//  Copyright © 2020 Alex Ivanova. All rights reserved.
+//
+
+import UIKit
+
+/// Calculate the frame of squares inside a view
+class Position {
+  private var safeArea: UIEdgeInsets
+  private var height: CGFloat
+  private var width: CGFloat
+  private var isLandscape: Bool
+  private var gapLength: CGFloat = 0
+  private var allGaps: CGFloat
+  private var countInRow: CGFloat = 0
+  private var length: CGFloat = 0
+  private var allSquares: CGFloat
+  private var offset: CGFloat
+  private var marginX: CGFloat = 0
+  private var marginY: CGFloat = 0
+  private var numberSquares: Int = 0
+  
+  var margin: CGFloat {
+    return isLandscape ? marginX : marginY
+  }
+  
+  init(parentView: UIView) {
+    safeArea = (parentView.superview?.safeAreaInsets)!
+    height = (parentView.superview?.bounds.height)! - safeArea.top - safeArea.bottom
+    width = (parentView.superview?.bounds.width)! - safeArea.right - safeArea.left
```

Comment:
Maybe you can first unwrap parentView.superview to avoid using `!`

Reply:
if I do not force unwrap I get an error message: "Return from initializer without initializing all stored properties"

Reply:
You need to tell the compiler that the initialization can fail, hence returns nil:
```
init?(parent: UIView)
```